Graphics/GraphicsEngineVulkan/src/VulkanUtilities/VulkanLogicalDevice.cpp

```diff
@@ -31,14 +31,495 @@
 #include "VulkanUtilities/VulkanDebug.hpp"
 #include "VulkanUtilities/VulkanObjectWrappers.hpp"
 
+// for KHR ray tracing emulation
+#include <mutex>
+#include <unordered_map>
+
 namespace VulkanUtilities
 {
 
-std::shared_ptr<VulkanLogicalDevice> VulkanLogicalDevice::Create(VkPhysicalDevice             vkPhysicalDevice,
+// KHR ray tracing emulation.
```

Comment:
Redirection functions should be moved to a separate file (RayTracingKHRviaNV.cpp or something), they do not belong to VulkanLogicalDevice.

Reply:
ok